src/pdf.rs

```diff
@@ -29,10 +29,10 @@ pub fn get_versions() -> Vec<PdfVersion> {
     vers_slice.iter().map(|v| PdfVersion::from(*v)).collect()
 }
 
-pub fn version_to_string(version: PdfVersion) -> Option<String> {
+pub fn version_to_string(version: PdfVersion) -> Option<&'static str> {
     unsafe {
         let res = ffi::cairo_pdf_version_to_string(version.into());
-        res.as_ref().and_then(|cstr| CStr::from_ptr(cstr as _).to_str().ok()).map(String::from)
+        res.as_ref().and_then(|cstr| CStr::from_ptr(cstr as _).to_str().ok())
```

Comment:
It [does](https://github.molgen.mpg.de/git-mirror/cairo/blob/master/src/cairo-pdf-surface.c#L666) return `NULL` if the `version` argument is not one of the defined values, which the bindings in `enum.rs` generally don't protect against (ie. there's the `__Unknown(i32)` variant). I suppose the enum could be made exhaustive and there could be an assert in it's from From<ffi> implementation. Then this could simplified. I don't have a strong feeling one way or the other, feel free to pick whichever you think is right...

Reply:
It even mentions in the docs of that function that it can return `NULL`, so this is fine then.